ipie/propagation/free_propagation.py

```diff
@@ -0,0 +1,92 @@
+import time
+
+import numpy
+
+from ipie.propagation.phaseless_base import (
+    construct_mean_field_shift,
+    construct_one_body_propagator,
+)
+from ipie.propagation.phaseless_generic import PhaselessGeneric
+from ipie.utils.backend import arraylib as xp
+from ipie.utils.backend import synchronize
+
+
+class FreePropagation(PhaselessGeneric):
+    """fp-afqmc propagator"""
+
+    def __init__(
+        self, time_step: float, exp_nmax: int = 6, verbose: bool = False, ene_0: float = 0.0
+    ) -> None:
+        super().__init__(time_step, exp_nmax=exp_nmax, verbose=verbose)
+        self.e_shift = ene_0  # unlike the dynamic shift in phaseless, this is a constant shift
+
+    def build(self, hamiltonian, trial=None, walkers=None, mpi_handler=None, verbose=False):
+        # dt/2 one-body propagator
+        start = time.time()
+        self.mf_shift = construct_mean_field_shift(hamiltonian, trial)
+        if verbose:
+            print(f"# Time to mean field shift: {time.time() - start} s")
+            print(
+                "# Absolute value of maximum component of mean field shift: "
+                "{:13.8e}.".format(numpy.max(numpy.abs(self.mf_shift)))
+            )
+        # construct one-body propagator
+        self.expH1 = construct_one_body_propagator(hamiltonian, self.mf_shift, self.dt)
+
+        # # Allocate force bias (we don't need to do this here - it will be allocated when it is needed)
+        self.vbias = None
+        # self.vbias = numpy.zeros((walkers.nwalkers, hamiltonian.nfields),
+        #                         dtype=numpy.complex128)
+        self.e_shift_1 = -hamiltonian.ecore - xp.sum(self.mf_shift**2) / 2.0
```

Comment:
I'm curious if this matters as it's just a constant in the numerator and denominator.

Reply:
the numerator and denominator samples will blow up individually without this, which might be problematic if using single precision e.g.